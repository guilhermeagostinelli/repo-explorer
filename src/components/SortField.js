import React from "react";
import Label from "./Label";

const SortField = ({
  valueNamePairs,
  sortByValue,
  sortTypeValue,
  onChangeSortByCb,
  onChangeSortTypeCb
}) => (
  <Label>
    Order by:
    <select
      value={sortByValue}
      onChange={e => onChangeSortByCb(e.target.value)}
    >
      {valueNamePairs.map(({ value, name }) => (
        <option key={value} value={value}>
          {name}
        </option>
      ))}
    </select>
    <select
      value={sortTypeValue}
      onChange={e => onChangeSortTypeCb(e.target.value)}
    >
      <option value="asc">Ascending order</option>
      <option value="desc">Descending order</option>
    </select>
  </Label>
);

export default SortField;
