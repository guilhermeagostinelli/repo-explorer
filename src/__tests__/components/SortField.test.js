import React from "react";
import { create } from "react-test-renderer";
import SortField from "../../components/SortField";
import "jest-styled-components";

describe("<SortField />", () => {
  const baseProps = {
    valueNamePairs: [
      { value: "name", name: "Name" },
      { value: "description", name: "Description" },
      { value: "created_at", name: "Creation date" },
      { value: "language", name: "Language" },
      { value: "stargazers_count", name: "Stars" }
    ],
    sortByValue: "name",
    sortTypeValue: "asc"
  };

  it("should match the snapshot", () => {
    const props = {
      ...baseProps,
      onChangeSortByCb: () => {},
      onChangeSortTypeCb: () => {}
    };

    const sortField = create(<SortField {...props} />);
    expect(sortField.toJSON()).toMatchSnapshot();
  });

  it("should fire the callback functions when the user changes the value of each of the select tags", () => {
    const handleChangeSortBy = jest.fn();
    const handleChangeSortType = jest.fn();
    const props = {
      ...baseProps,
      onChangeSortByCb: handleChangeSortBy,
      onChangeSortTypeCb: handleChangeSortType
    };
    const sortField = create(<SortField {...props} />);
    const instance = sortField.root;
    const selects = instance.findAllByType("select");
    selects[0].props.onChange({ target: { value: "description" } });
    expect(handleChangeSortBy).toHaveBeenCalledTimes(1);
    expect(handleChangeSortBy).toHaveBeenCalledWith("description");
    selects[1].props.onChange({ target: { value: "desc" } });
    expect(handleChangeSortType).toHaveBeenCalledTimes(1);
    expect(handleChangeSortType).toHaveBeenCalledWith("desc");
  });
});
