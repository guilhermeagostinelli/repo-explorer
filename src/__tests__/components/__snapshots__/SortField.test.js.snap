// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortField /> should match the snapshot 1`] = `
.c0 > input,
.c0 > select {
  margin-left: 10px;
}

<label
  className="c0"
>
  Order by:
  <select
    onChange={[Function]}
    value="name"
  >
    <option
      value="name"
    >
      Name
    </option>
    <option
      value="description"
    >
      Description
    </option>
    <option
      value="created_at"
    >
      Creation date
    </option>
    <option
      value="language"
    >
      Language
    </option>
    <option
      value="stargazers_count"
    >
      Stars
    </option>
  </select>
  <select
    onChange={[Function]}
    value="asc"
  >
    <option
      value="asc"
    >
      Ascending order
    </option>
    <option
      value="desc"
    >
      Descending order
    </option>
  </select>
</label>
`;
